ostd/src/task/mod.rs

```diff
@@ -55,6 +56,10 @@ pub struct Task {
     #[expect(dead_code)]
     kstack: KernelStack,
 
+    /// This is to check if we would context switch to a running task.
+    // TODO: use `#[cfg(debug_assertions)]` when the scheduler API is sound.
+    foreground: AtomicBool,
```

Comment:
I don't think `foreground` is a good word. This will make people wonder what are foreground tasks and what are background ones. In the POSIX worlds, they are related to job control. By naming this field to `foreground`, we are introducing these potentially confusing concepts into the readers' mind.

I suggest renaming the field and revising its comment as follows.

```
/// Indicates whether it is safe to switch to this task.
///
/// This flag is used to guard against the memory unsafe situation
/// where a single task is scheduled to run on two CPUs simultaneously.
/// This may happen if the injected task scheduler is buggy.
safe_to_switch: AtomicBool,
```